src/Products/ProdottoVenduto.java

```diff
@@ -0,0 +1,93 @@
+package Products;
+import Enums.TipoElettronico;
+import Utility.ProdottoVendutoReader;
+import java.util.List;
+
+import java.io.FileNotFoundException;
+
+public class ProdottoVenduto {
+    private TipoElettronico tipoProdotto;
+    private int quantitaVenduta;
+    private double spesaTotale;
+    private static String stampa1 = "La media dei Smartphone venduti è:";
+    private static String stampa2 = "La media dei Tablet venduti è:";
+    private static String stampa3 = "La media dei Notebook venduti è:";
+
+    public ProdottoVenduto(TipoElettronico tipoProdotto, double spesaTotale, int quantitaVenduta) {
+        this.tipoProdotto = tipoProdotto;
+        this.spesaTotale = spesaTotale;
+        this.quantitaVenduta = quantitaVenduta;
+    }
+
+    public ProdottoVenduto(){};
+
+    public TipoElettronico getTipoProdotto() {
+        return tipoProdotto;
+    }
+
+    public void setTipoProdotto(TipoElettronico tipoProdotto) {
+        this.tipoProdotto = tipoProdotto;
+    }
+
+    public int getQuantitaVenduta() {
+        return quantitaVenduta;
+    }
+
+    public void setQuantitaVenduta(int quantitaVenduta) {
+        this.quantitaVenduta = quantitaVenduta;
+    }
+
+    public double getSpesaTotale() {
+        return spesaTotale;
+    }
+
+    public void setSpesaTotale(double spesaTotale) {
+        this.spesaTotale = spesaTotale;
+    }
+
+    private double calcoloSpesaMediaSmartphone() throws FileNotFoundException {
```

Comment:
non servono 3 metodi diversi, il valore dell'enum da considerare e da usare come filtro per i prodotti si può passare come parametro del metodo.

Reply:
Ho sostituito i tre metodi con un singolo metodo che calcola la spesa in base al tipo